### ISSUE TITLE
Consul DNS SRV and gRPC plugin

### ISSUE THREAD

**Issue by Googlom** [luraproject/lura#370]
Hi! I'm trying to couple consul's srv record and grpc plugin([https://github.com/devopsfaith/krakend-grpc](url)).
My endpoint config:
```json
...
{
    "endpoint": "/test",
    "method": "GET",
    "backend": [
        {
            "host": [ "testsrv.service.dc1.consul" ],
            "url_pattern": "test.TestSrv",
            "method": "Hello",
            "encoding": "grpc-test.TestSrv.Hello",
            "disable_host_sanitize": true,
            "sd": "dns"
        }
    ]
}
...
```
When calling the endpoint I'm getting error:
`Error #01: parse "http://testnode.node.dc1.consul.:23312test.TestSrv": invalid port ":23312test.TestSrv" after host`

And if I change `"url_pattern"` to `"/test.TestSrv"` and call endpoint:
`Error #01: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp: address testsrv.service.dc1.consul: missing port in address"`

Please, show me where to dig :-D

**Comment by kpacha** [comment]
Did you check your custom build with the same environment provided at https://github.com/devopsfaith/krakend-grpc/tree/master/example? The fewer moving parts the better. Then you can move to consume your custom grpc services and later introduce a service discovery

**Comment by Googlom** [comment]
> Did you check your custom build with the same environment provided at https://github.com/devopsfaith/krakend-grpc/tree/master/example? 

That example needs to be fixed, because it doesn't even compile because of wrong imports...

> The fewer moving parts the better. Then you can move to consume your custom grpc services and later introduce a service discovery

Ok, I'll try with minimal setup.

**Comment by Googlom** [comment]
Well, I've fixed that example and make it compile and work properly. So
1) That example with no configuration change (backend in **localhost:50051** and **helloworld** service) -> Works fine. OK.
2) Changed **helloworld** service to my custom service, ip and port still static -> OK. All grpc methods serving properly.
3) Registered my custom service with consul and changed backend host config to `"testsrv.service.dc1.consul:22222"` -> still OK
4) Added `"sd": "dns"` to backend config, placed `dnssrv.Register()` in main function of krakend. Host config is still static (`"testsrv.service.dc1.consul:22222"`) -> OK
5) Set host config to `"testsrv.service.dc1.consul"` -> Error:
`Error #01: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp: address testsrv.service.dc1.consul: missing port in address"`

**Comment by Googlom** [comment]
Hi! I have a question about resolving in **dnssrv** package.
https://github.com/devopsfaith/krakend/blob/3c72ecff447c764010ca725f854c3896818fe011/sd/dnssrv/subscriber.go#L87-L92

Why http:// prefix added to host:port in name resolution level? Why not return host:port, then in backend level decide what to do with this host:port?
I haven't looked yes the whole codebase, just **sd** package. So I don't know if this is design purpose..

**Comment by alombarte** [comment]
Thanks for your suggestion @Googlom .

We will fix this on #667
